src/Form/index.tsx

```diff
@@ -593,27 +593,34 @@ function Form({
   const getNewStep = async (newKey: any) => {
     // @ts-expect-error TS(2531): Object is possibly 'null'.
     let newStep = steps[newKey];
-    while (true) {
-      const loadCond = newStep.next_conditions.find((cond: any) => {
-        if (cond.element_type !== 'step') return false;
-        const notAuth =
-          cond.rules.find((r: any) => r.comparison === 'not_authenticated') &&
-          !initState.authId &&
-          // Re: firebaseConfirmationResult, the user hasn't authenticated yet
-          // but they're in the process of doing so and we don't want to
-          // consider them "unauthenticated" for the purposes of redirecting
-          !window.firebaseConfirmationResult;
-        const auth =
-          cond.rules.find((r: any) => r.comparison === 'authenticated') &&
-          initState.authId;
-        return notAuth || auth;
-      });
-      if (loadCond) {
-        if (changeStep(loadCond.next_step_key, newKey, steps, history)) {
-          clearLoaders(true);
-          return;
-        }
-      } else break;
+
+    const metadata =
+      integrations.stytch?.metadata ?? integrations.firebase?.metadata;
+    if (metadata?.auth_gate_steps?.length > 0) {
+      const userAuthed = Boolean(initInfo().authId);
+      const nextStepIsProtected = metadata.auth_gate_steps.includes(newStep.id);
+      const findStep = (stepId: string) =>
+        // @ts-expect-error TS(2531): Object is possibly 'null'.
+        Object.values(steps).find((step) => step.id === stepId).key;
+      let nextStep: undefined | string;
+
+      if (userAuthed && !nextStepIsProtected)
+        nextStep = findStep(metadata.login_step);
+      else if (
+        !userAuthed &&
+        // Re: firebaseConfirmationResult, the user hasn't authenticated yet
+        // but they're in the process of doing so and we don't want to
+        // consider them "unauthenticated" for the purposes of redirecting
+        !window.firebaseConfirmationResult &&
```

Comment:
i think we can remove this tbh, it's a weird case that probably doesn't need to be supported